compiler/qsc_doc_gen/src/generate_docs.rs

```diff
@@ -22,6 +22,135 @@ use std::sync::Arc;
 type Files = Vec<(Arc<str>, Arc<str>, Arc<str>)>;
 type FilesWithMetadata = Vec<(Arc<str>, Arc<Metadata>, Arc<str>)>;
 
+type ToC = FxHashMap<Rc<str>, Vec<Arc<Metadata>>>;
+
+struct Metadata {
+    uid: String,
+    title: String,
+    topic: String,
+    kind: MetadataKind,
+    package: PackageKind,
+    namespace: Rc<str>,
+    name: Rc<str>,
+    summary: String,
+    signature: String,
+}
+
+impl Metadata {
+    fn fully_qualified_name(&self) -> String {
+        let mut buf = if let PackageKind::AliasedPackage(ref package_alias) = self.package {
+            vec![format!("{package_alias}")]
+        } else {
+            vec![]
+        };
+
+        buf.push(self.namespace.to_string());
+        buf.push(self.name.to_string());
+        buf.join(".")
+    }
+
+    fn display_for_toc(&self) -> String {
+        format!(
+            "---
+uid: {}
+title: {}
+description: {}
+author: bradben
```

Comment:
I suggest we move this to some pipeline or deployment script.

Reply:
All of these have been replaced with `{AUTHOR}` and `{MS_AUTHOR}` tags to be replaced during publishing.